src/main/collectors/InstantSearchQueryCollector.js

```diff
@@ -27,7 +29,8 @@ class InstantSearchQueryCollector {
    * @param {object} writer - The writer to send the data to
    */
   attach(writer) {
-    var searchBox = document.querySelector(this.searchFieldSelector);
+    var doc = this.getContext() ? this.getContext().getDocument() : document;
```

Comment:
is it a good idea to introduce helper methods in AbstractCollector:
```
getDocument() {
  return this.getContext() ? this.getContext().getDocument() : document;
} 
```

and similar for `Window`.
This way the specializations can just use `const doc = getDocument();`

Reply:
Good point, adjusted